src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -393,6 +400,25 @@ private void ClaimPeerTip(int networkPeerId, uint256 tipHash)
             peersClaimingThisHeader.Add(networkPeerId);
         }
 
+        /// <inheritdoc />
+        public void ClaimPeerTipForMiner(uint256 tipHash)
+        {
+            HashSet<int> peersClaimingThisHeader;
```

Comment:
I suggest just call the existing method `ClaimPeerTip` then at the bottom of this method find the last MinerPeerId on the same branch and remove it.

Though I am not sure if w should remove it at all, we could always keep mined branches of blocks.

@noescape00 please review this fix as well (and maybe also @Aprogiena if available )

The reason we need to mark a mined block with its own special peer is that it does not get reorged away in case of a longer chain is presented and the node jumps to that chain.